R/taxize_nexml.R

```diff
@@ -6,38 +6,47 @@
 #' containing the corresponding identifier. 
 #' @param nexml a nexml object
 #' @param type the name of the identifier to use
-#' @param ... additional arguments (not implemented yet)
+#' @param ... additional arguments to `[taxald::get_ids()]`
 #' @export 
 #' @examples \dontrun{
 #' data(bird.orders)
 #' birds <- add_trees(bird.orders)
 #' birds <- taxize_nexml(birds, "NCBI")
 #' }
 # @importFrom taxize get_uid
-taxize_nexml <- function(nexml, type = c("NCBI"), ...){
+taxize_nexml <- function(nexml, type = c("ncbi", "itis", "col", "tpl",
+                                         "gbif", "wd"), ...){
   
   ## Soft dependency on taxize
-  if (!requireNamespace("taxize", quietly = TRUE)) {
-    stop("taxize package required to convert look up taxonomic ids",
+  if (!requireNamespace("taxald", quietly = TRUE)) {
+    stop("taxald package required to convert look up taxonomic ids",
          call. = FALSE)
   }
-  get_uid <- getExportedValue("taxize", "get_uid")
+  get_ids <- getExportedValue("taxald", "get_ids")
```

Comment:
yes, but this change is intentional -- `taxize::get_uid` is the NCBI-specific function (though you probably wouldn't guess that from the name), while `taxize::get_ids` is a wrapper around a suite of potential authorities (including `itis`, `col`, and `gbif`.  The original RNeXML implementation only supported NCBI by hardwiring `get_uid`, while at the same time promising to support future versions (we've always had a `type` argument, but only supporting `type="NCBI"`.  So this PR delivers on the support for some alternate authorities and changes the name to match the corresponding `taxize` function.  

(Side note, but `taxize::get_ids()` did nothing to standardize the return object provided by wrapping the different authorities, so you couldn't just swap this back to `taxize::get_ids()`.  `taxald` mimics the input, but always provides a consistent output regardless of the authority source).

Reply:
In fact, `get_ids` only gets used a single time. Is there a good reason to not simply replace
```R
    taxa_ids <- get_ids(labels, type, format = "uri", ...)
```
with
```R
    taxa_ids <- taxald::get_ids(labels, type, format = "uri", ...)
```

Reply:
But I'm nitpicking here 😄 so feel free to just ignore.

Reply:
So I am intentionally avoiding the use of an explicit reference to the package (e.g. `taxald::`) because that would cause R CMD CHECK to throw a WARNING, because `taxald` is only suggested and not imported.  Importing from a function from another package using `getExportedValue` is something of an accepted hack to allow a package to have "soft" dependencies.  

We did this with `taxize` initially because `taxize` is a huge dependency with lots of additional dependencies and listing it in Imports would make it required for a user who really just wants to do simple parsing of NeXML files and doesn't need this 'bonus' functionality.  `taxald` is a bit lighter, but still not 'core' functionality; and besides, we can't have it in Imports if `taxald` isn't on CRAN, though we can have it in Suggests.